src/sentry/buffer/redis.py

```diff
@@ -194,10 +200,10 @@ def incr(self, model, columns, filters, extra=None, signal_only=None, return_inc
             # hook here
             # e.g. "update score if last_seen or times_seen is changed"
             for column, value in extra.items():
-                # TODO(dcramer): once this goes live in production, we can kill the pickle path
-                # (this is to ensure a zero downtime deploy where we can transition event processing)
-                pipe.hset(key, "e+" + column, pickle.dumps(value))
```

Comment:
I have a really vague memory of us trying to kill this pickle before and having it go badly, but I don't remember why.

Reply:
@wedamija going to address both your comments here. IIRC it was @beezz who used servicedelegator to migrate buffers between redis hosts, I didn't think this through but i was going to pick whatever strategy he chose there to do the same kind of migration off of redis blaster towards redis cluster. your plan to dual-*read* and then switch over writes sounds reasonable to me.

the json codepath at least succeeds in tests but i can imagine there is some buffer arg combination out there that does not pass through the json serializer but is pickleable. the only relevant commit i can find is https://github.com/getsentry/sentry/commit/38f65994444b98cb7c6d94029796fa93f6488b86, so perhaps it failed last time just due to the fact that deploys are not atomic and there was no migration to a new cluster, which should not be a problem here.

we'd probably want a percentage-based rollout anyway so we can catch any pickle errors and other problems, eg. regarding load.